pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -437,5 +440,62 @@ private void doShutDownBrokerGracefully(int maxConcurrentUnloadPerSec,
         pulsar().getBrokerService().unloadNamespaceBundlesGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
         pulsar().closeAsync();
     }
+
+    /**
+     * dynamically update logger level at runtime.
+     *
+     * @param targetClassName : class name to update
+     * @param targetLevel     : target log level
+     */
+    private CompletableFuture<Void> internalUpdateLoggerLevelAsync(String targetClassName, String targetLevel) {
+        CompletableFuture<Void> loggerLevelFuture = new CompletableFuture<>();
+        try {
+            String className;
+            // if set "ROOT" will take effect to rootLogger
+            if (targetClassName.trim().equalsIgnoreCase("ROOT")) {
+                className = LogManager.ROOT_LOGGER_NAME;
+            } else {
+                className = targetClassName;
+            }
+            Level newLevel;
+            try {
+                newLevel = Level.valueOf(targetLevel);
+                Configurator.setAllLevels(className, newLevel);
+            } catch (IllegalArgumentException | NullPointerException e) {
+                // Unknown Log Level or NULL
+                throw new RestException(Status.PRECONDITION_FAILED, "Invalid logger level.");
+            }
+        } catch (RestException re) {
+            throw re;
+        } catch (Exception ie) {
+            throw new RestException(Status.PRECONDITION_FAILED, "Internal error.");
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @POST
+    @Path("/logging/{classname}/{level}")
+    @ApiOperation(value =
+      "dynamic update logger level at runtime. This operation requires Pulsar super-user privileges.")
+    @ApiResponses(value = {
+      @ApiResponse(code = 204, message = "class logger level updated successfully"),
+      @ApiResponse(code = 403, message = "You don't have admin permission to update logger level."),
+      @ApiResponse(code = 500, message = "Internal server error")})
+    public void updateLoggerLevelDynamically(@Suspended final AsyncResponse asyncResponse,
+                                             @PathParam("classname") String classname,
```

Comment:
It should be the logger name here, not the class name. Usually, we will use the class name as the logger name, but not 100%